yktrace/src/sir.rs

```diff
@@ -64,7 +65,9 @@ lazy_static! {
 
 impl Sir {
     pub fn read_file(file: &Path) -> Result<Sir, ()> {
-        let ef = elf::File::open_path(file).unwrap();
+        // SAFETY: Not really, we hope that nobody changes the file underneath our feet.
+        let data = unsafe { Mmap::map(&File::open(file).unwrap()).unwrap() };
```

Comment:
So mmap is faster?

Reply:
There are two options when using `object`. Either you load the whole file into a `Vec<u8>` or you mmap the file. The former would also load the `.text` and `.data` sections, not just the `.yksir_*` sections. Rustc also uses mmap for metadata loading. I have not benchmarked both approaches though.